bitbots_ball_filter/src/bitbots_ball_filter/ball_filter.py

```diff
@@ -107,10 +107,55 @@ def _dynamic_reconfigure_callback(self, config, level):
         self.filter_timer = rospy.Timer(rospy.Duration(self.filter_time_step), self.filter_step)
         return config
 
+    def ball_callback2(self, msg: PoseWithCertaintyArray, optional_distance = True):
+        """handles incoming ball messages"""
+        if optional_distance:
+            if msg.poses:
+                balls = [ball for ball in msg.poses if ball.confidence > self.min_ball_confidence]#only look at balls who have at least some minimal certainty
+                
+                if not balls: #if there are not any balls with a high enough confidence
+                    return
+                for ball in balls:
+                    last_distance = 5 #we need a number that is rather hihg, I do not know how to chose that properly
+                    if(self.calc_distance_between_balls(ball, msg.header)< last_distance):
+                        last_distance = self.calc_distance_between_balls(ball, msg.header)
+                        self.last_ball_msg = ball
+                        ball_buffer = PointStamped()
+                        ball_buffer.header = msg.header
+                        ball_buffer.point = ball.pose.pose.position
```

Comment:
Oh. Maybe it would be best to create a list of tuples (transformed_ball, distance) and then sorting the list by the distance. Afterwards, you can select the first/last element of the sorted list.